src/main/java/software/bernie/techarium/block/base/MachineBlockRotationXYZ.java

```diff
@@ -0,0 +1,32 @@
+package software.bernie.techarium.block.base;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.state.DirectionProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.Direction;
+import software.bernie.techarium.tile.base.MachineTileBase;
+import software.bernie.techarium.trait.block.BlockBehaviour;
+
+public class MachineBlockRotationXYZ<T extends MachineTileBase> extends MachineBlock<T> {
```

Comment:
So the whole point of the Trait system is to avoid the [diamond inheritance problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem), i.e. making a ton of subclasses and then having to make specific combinations because java doesn't support multiple inheritance in classes. The Trait system is essentially the same idea as interfaces except traits can have fields and are easily composable.

In this specific case what you should do instead is:
- Create a [Trait](https://github.com/bernie-g/Techarium/blob/master/src/main/java/software/bernie/techarium/trait/block/BlockTraits.java) called something like `BlockRotationTrait`, and then have several enums or some way of storing which rotation types (and add that enum as a constructor parameter). Then, have that trait have several methods like `getStateForPlacement`, `createBlockStateDefinition`, etc. that all handle correctly based on the direction enum in that trait. Then, remove these two MachineBlockRotation classes and override the vanilla blockstate methods in `TechariumBlock` and redirect them to the Trait's methods. That way we don't have to override these MachineBlockRotation classes.